motion_predict/src/predict_ctrv.cpp

```diff
@@ -45,25 +45,30 @@ std::tuple<double, double> localVelOrientationAndMagnitude(const double v_x, con
   }
   else
   {
-    local_v_orientation = asin(v_y / v_mag);
+    local_v_orientation = atan2(v_y, v_x);
   }
   return std::make_tuple(local_v_orientation, v_mag);
 }
 
 CTRV_State buildCTRVState(const geometry_msgs::msg::Pose& pose, const geometry_msgs::msg::Twist& twist)
 {
-  geometry_msgs::msg::Quaternion quat = pose.orientation;
-  Eigen::Quaternionf e_quat(quat.w, quat.x, quat.y, quat.z);
-  Eigen::Vector3f rpy = e_quat.toRotationMatrix().eulerAngles(0, 1, 2);
-
+  // Need a logic here to possible detect whether if twist.linear.x,y
+  // is in map frame. https://github.com/usdot-fhwa-stol/carma-platform/issues/2407
   auto vel_angle_and_mag = localVelOrientationAndMagnitude(twist.linear.x, twist.linear.y);
 
   CTRV_State state;
   state.x = pose.position.x;
   state.y = pose.position.y;
-  state.yaw =
-      rpy[2] + std::get<0>(vel_angle_and_mag);  // The yaw is relative to the velocity vector so take the heading and
-                                                // add it to the angle of the velocity vector in the local frame
+  state.yaw = std::get<0>(vel_angle_and_mag); // Currently, object's linear velocity is already in map frame.
+                                              // Orientation of the object cannot be trusted for objects
+                                              // as it could be drifting sideways while facing other direction
+
+  // geometry_msgs::msg::Quaternion quat = pose.orientation;
```

Comment:
I think we can remove this commented out code now, with the assumption that this package assumes inputs in the `map` frame.